docs/DeveloperGuide.md

```diff
@@ -2,12 +2,153 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
+
+[Original Source](https://github.com/nus-cs2113-AY2425S1/tp)
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## Overall Class Diagram
+![overallClass.jpg](./Diagrams/overallClass.jpg)
+On a high level, whenever Wiagi is started, it will load SpendingList and IncomeList from Storage if it exists, else, 
+new lists would be created
+Wiagi then takes in user input via the UI class, then parse and executes the command through the parser class.
+The related output is printed through the UI class.
+At the end of the run, or when the user exits the application, Wiagi will save the lists.
+
+### Storage
+To load data from previous session:
+Within Wiagi constructor, Storage class is constructed, which will load and initialise incomes, spendings and
+password by de-serialising the text at their distinct file paths. Wiagi will then initialise it incomes and spendings
+based on the member in the Storage class.
+![storageLoad.png](./Diagrams/storageLoad.png)
```

Comment:
Consider to adjust the background colour since the lifelines appear to be invisible.
<img width="925" alt="Screenshot 2024-10-30 at 12 36 53" src="https://github.com/user-attachments/assets/aa80db85-ab3f-402c-8062-30ec181a03b7">